datadog/fwprovider/resource_datadog_app.go

```diff
@@ -0,0 +1,292 @@
+	var plan appResourceModel
+	diags := request.Plan.Get(ctx, &plan)
+	response.Diagnostics.Append(diags...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	id, err := uuid.Parse(plan.ID.ValueString())
+	if err != nil {
+		response.Diagnostics.AddError("Error parsing id as uuid", err.Error())
+		return
+	}
+
+	updateRequest, err := appModelToUpdateApiRequest(plan)
+	if err != nil {
+		response.Diagnostics.AddError("Error building update app request", err.Error())
+		return
+	}
+
+	_, httpResp, err := r.Api.UpdateApp(r.Auth, id, *updateRequest)
+	if err != nil {
+		if httpResp != nil {
+			// error body may have useful info for the user
+			body, err := io.ReadAll(httpResp.Body)
+			if err != nil {
+				response.Diagnostics.AddError("Error reading error response", err.Error())
+				return
+			}
+			response.Diagnostics.AddError("Error updating app", string(body))
+		} else {
+			response.Diagnostics.AddError("Error updating app", err.Error())
+		}
+		return
+	}
+
+	// Save data into Terraform state
+	diags = response.State.Set(ctx, &plan)
+	response.Diagnostics.Append(diags...)
+}
+
+func (r *appResource) Delete(ctx context.Context, request resource.DeleteRequest, response *resource.DeleteResponse) {
+	var state appResourceModel
+	diags := request.State.Get(ctx, &state)
+	response.Diagnostics.Append(diags...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	id, err := uuid.Parse(state.ID.ValueString())
+	if err != nil {
+		response.Diagnostics.AddError("Error parsing id as uuid", err.Error())
+		return
+	}
+	_, httpResp, err := r.Api.DeleteApp(r.Auth, id)
+	if err != nil {
+		if httpResp != nil {
+			// error body may have useful info for the user
+			body, err := io.ReadAll(httpResp.Body)
+			if err != nil {
+				response.Diagnostics.AddError("Error reading error response", err.Error())
+				return
+			}
+			response.Diagnostics.AddError("Error deleting app", string(body))
+		} else {
+			response.Diagnostics.AddError("Error deleting app", err.Error())
+		}
+		return
+	}
+}
+
+func apiResponseToAppModel(resp datadogV2.GetAppResponse) (*appResourceModel, error) {
+	appModel := &appResourceModel{
+		ID: types.StringValue(resp.Data.GetId().String()),
+	}
+
+	data := resp.GetData()
+	attributes := data.GetAttributes()
+
+	marshalledBytes, err := json.Marshal(attributes)
+	if err != nil {
+		err = fmt.Errorf("error marshaling attributes: %s", err)
+		return nil, err
+	}
+	appModel.AppJson = jsontypes.NewNormalizedValue(string(marshalledBytes))
+
+	return appModel, nil
+}
+
+func appModelToCreateApiRequest(appModel appResourceModel) (*datadogV2.CreateAppRequest, error) {
+	attributes := datadogV2.NewCreateAppRequestDataAttributesWithDefaults()
+
+	// decode encoded json into string and then decode that into the attributes struct
+	var appJsonString string
+	err := json.Unmarshal([]byte(appModel.AppJson.String()), &appJsonString)
```

Comment:
In providers, you should use `ValueString()` to get a string instead of `String()` which is for debugging. when i used `String()` it made the strings all double-quoted

this comment applies to all usage of `String()` in this PR